### DISCUSSION TITLE
Port does not match.

### DISCUSSION THREAD

**Discussion by CtrlAltSpoods** [dhruvinsh/ws-ephemeral#3] [General]
Hi all,

Given that this tool just automatically resets Windscribe's port forward page and gets us a newly assigned port, it does not appear to work.
qBittorrent is not smart enough to detect that port redirection is in place, and it wants a matching port.

I believe this project will need some more work to actually bring a real world advantage, as someone else pointed out you must change your connection method to **JUST _TCP_** if you want your firewall status to be accurate. (**uTP** will make it show green **even if it's blocked**)

I think this tool will need to control qBittorrent's config file to update it with a matching port provided by Windscribe if it is to work correctly.

If I have this wrong, or you think there is a reason this could work, by all means please let me know.

Kind regards

**Comment by dhruvinsh** [comment]
Hi,
Purpose of this tool is to work with ephemeral ports and request one specific port which you have already configured on qbit.

What i suggest is, set `Port used for incoming connections` under qbit to something like 12001. And with help of this tool you will always requests the same port (that is 12001) on WS service. That way there is no change required on qbit and WS will always assign you 12001 port for your VPN channel. WS on its own would take care of the internal mapping which we don't need to worry about.

**Comment by x134h34kr** [comment]
yeah, this doesn't work if you use qBitTorrent with a port that doesn't match. Tested both scenarios and only matching ports work.

**Reply by dhruvinsh** [reply]
Not sure why it behaving in your case. But for me it almost always work. One time I had a issue but turns out wireguard config needed an update on my end. on WS I always use random port making but I always ask for one specific port.

Also, I am hopefully by end of week (depending on time) I will push some update that I am using for myself. With that you will be able to setup matching port. And automated qbit setting as well. I agree that configureing qbit is not that great!

**Reply by x134h34kr** [reply]
Cool, looking forward to an update for matching ports and auto qbit configuration. Appreciate your work on this.

**Reply by dhruvinsh** [reply]
pushed the update. try it out and let me know

**Comment by CtrlAltSpoods** [comment]
The torrent client doesn't change ports based on Windscribe though and it
advertises in the trackers the wrong port that is not open.

This makes other peers try to connect to said port which is not open.

Even if they did try to connect to the 'ephemeral' port, the torrent client
doesn't listen on all ports so it will ignore the connection attempt.

I believe if you have the utp protocol enabled then your firewall status
icon will be green which misleads you into thinking it's working properly.

**Reply by CtrlAltSpoods** [reply]
Replying via email is not a good experience lol.
It has removed all my formatting -_- thankyou GitHub.

**Reply by dhruvinsh** [reply]
Apologies for my ignorance. I did check my config and just today I noticed that I had "tcp & utp" set. But it seems that everytime port forwarding get setup it automatically picked same matching port for me, at least from last 2 times I can see that.

I am working on newer version that should deal with this issue.
